### ISSUE TITLE
if NamePipeClientStream pipename have “~” can't connect

### ISSUE THREAD

**Issue by heweisheng** [dotnet/runtime#102807]
### Description

if pipeName have "~" such as "1~(2)" can't connect ,FileStream can't connect too

### Reproduction Steps

test code:
```           string pipename = "1~(2)";
            NamedPipeServerStream serverStream = new NamedPipeServerStream(pipename, PipeDirection.InOut);

            NamedPipeClientStream pipeClient = new NamedPipeClientStream(".", pipename);
            pipeClient.Connect();
            Console.WriteLine("Hello, World!");
```
if pipename is "1~(2)" client can't connect
if pipename is "1(2)" client connect

### Expected behavior

![image](https://github.com/dotnet/runtime/assets/37769332/4d83280c-36d6-4c65-863d-12a4ee6485d0)

### Actual behavior

![image](https://github.com/dotnet/runtime/assets/37769332/694b823f-3ed1-45bf-b2ae-d430a6984838)

### Configuration

os: windows 11 x64
.netcore 8.06

**Comment by heweisheng** [comment]
the pipename test in win32 CreateFile and CreateNamedPipe api is ok why?

**Comment by Jozkee** [comment]
I don't think we treat `~` in any special way in System.IO.Pipes APIs so this should come from Windows. This requires further investigation to be actionable.

Tagging @JeremyKuhne if by luck he knows about this.

**Comment by JeremyKuhne** [comment]
Any character other than backslash _should_ be ok. https://learn.microsoft.com/windows/win32/ipc/pipe-names

I don't see any special handling in the .NET code, but I do get the failure as well.  https://github.com/dotnet/runtime/blob/7f7bb7b1530b85ec8d0d4b6e5f5429a6b729b076/src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs#L139

We'll probably have to defer to Windows on why this character isn't working.

**Comment by danmoseley** [comment]
Does it fail on .NET Framework too?
